drivers/serial/uart_pl011.c

```diff
@@ -423,6 +429,9 @@ static struct uart_device_config pl011_cfg_port_0 = {
 
 static struct pl011_data pl011_data_port_0 = {
 	.baud_rate = DT_PL011_PORT0_BAUD_RATE,
+#if defined(CONFIG_UART_PL011_SHARED_IRQ)
+	.shared_irq_dev_name = DT_INST_0_SHARED_IRQ_LABEL,
```

Comment:
Where or how is DT_INST_0_SHARED_IRQ_LABEL getting set?

Reply:
That's generated automatically when the DT entry for the shared IRQ is parsed. For example from the DT entry:
```
sharedirq0: sharedirq0 {                               
        compatible = "shared-irq";                    
        label = "SHARED_IRQ";                      
        interrupts = <GIC_SPI 1 0 IRQ_TYPE_LEVEL>;       
        interrupt-controller;      
        #interrupt-cells = <0>;                         
        status = "okay";       
};
```
`DT_INST_0_SHARED_IRQ_LABEL="SHARED_IRQ"` is generated in `generated_dts_board.conf` at compile time.